src/rgw/rgw_lc.cc

```diff
@@ -2005,6 +2009,14 @@ int RGWLC::process(int index, int max_lock_secs, LCWorker* worker,
     if (entry.bucket.empty())
       goto exit;
 
+    /* skip over bucket if processing for only a single bucket was
+     * requested, and this one isn't it */
+    if (! bucket_name.empty()) {
+      if (! boost::algorithm::starts_with(entry.bucket, bucket_prefix)) {
+	goto next_bucket;
```

Comment:
could we expose a call directly to `bucket_lc_process()`, and avoid having to walk the LC list looking for this bucket?

is `RGWLC::process()` doing anything else that's important here? i see that it updates the `head.marker = entry.bucket` when it finds the specified bucket, but that might cause normal LC processing to skip past those earlier buckets as well

Reply:
expose:  yeah, maybe;  I'll look at it now I have index selection done